### ISSUE TITLE
Face parsing error

### ISSUE THREAD

**Issue by GoldenBrownie** [rauldlnx10/sd-webui-swap-mukham#9]
It works really well, and I actually like the outputs of this more than the google colab roop one. But that's until I enable the face parsing option. Then it gives the error 'NoneType' object has no attribute 'items'.
![image_2023-09-14_211032560](https://github.com/rauldlnx10/sd-webui-swap-mukham/assets/31779489/852363c8-0d0b-432a-b7f3-f6068eff929b)
It's possible I'm doing something wrong, but if that's the case I don't really know what.

Also, the download models button gives the error "UnboundLocalError: local variable 'filename' referenced before assignment". So those could be related? I downloaded the inswapper model manually but maybe the face parsing uses something different
![image_2023-09-14_211212198](https://github.com/rauldlnx10/sd-webui-swap-mukham/assets/31779489/560839c9-dd10-4896-8754-eb1edd70732d)

**Comment by rauldlnx10** [comment]
try the new version, has a lot new features, has a lot new features, let me know if it solves the error.

 have modified the code, trying to make it as similar to the original as possible.  

`git clone -b new https://github.com/rauldlnx10/sd-webui-swap-mukham.git`

**Comment by GoldenBrownie** [comment]
The download models button works now, which is awesome! The face parsing still gives the same error though. Personally that's not a huge deal, I just wanted to see the output difference with it on/off but just letting you know. Thanks for the update tho. It's an awesome extension to have

**Comment by rauldlnx10** [comment]
> The download models button works now, which is awesome! The face parsing still gives the same error though. Personally that's not a huge deal, I just wanted to see the output difference with it on/off but just letting you know. Thanks for the update tho. It's an awesome extension to have

Ill check it, but im not a programmer so will take time...  btw you can mask with the new version so face parsing is not really needed...

**Comment by GoldenBrownie** [comment]
Yeah I noticed but like I said not a big deal, just letting you know there is an error. I just wanted to see the difference with and without it. I'm not a programmer either and I certainly wouldn't have been able to do what you've done so take your time! I'm guessing it's possible the error has to do with my computer/settings/video or whatever, and not the extension itself. My computer has been dead for a few days now so I can't really test more. But I'll give it a try with a video with a clear face on it once I get it back.

**Comment by GoldenBrownie** [comment]
Tried again using a video with a very clear face and I get the same error. Also, I might be missing something but after messing around with the masks it seems to make no difference at all what is/isn't included. Given the way the Masks tab is presented, I wonder if they only work if face parsing is enabled?

**Comment by rauldlnx10** [comment]
> Tried again using a video with a very clear face and I get the same error. Also, I might be missing something but after messing around with the masks it seems to make no difference at all what is/isn't included. Given the way the Masks tab is presented, I wonder if they only work if face parsing is enabled?

Try enabling only Parse from target and adding the items you want to be swapped...

**Comment by GoldenBrownie** [comment]
Unless I'm missing something, there's no "parse from target" option. There's "enable face parsing", the items to be swapped, a few more settings, and then "Laplacian blending".

**Comment by rauldlnx10** [comment]
> Unless I'm missing something, there's no "parse from target" option. There's "enable face parsing", the items to be swapped, a few more settings, and then "Laplacian blending".

Download the new version, you must do:

git clone -b New https://github.com/rauldlnx10/sd-webui-swap-mukham.git

**Comment by Kagitoshi** [comment]
I had the same error so i used the new one and it just doesn't work at all.

I tried using my face with a picture of superman and just superman came out on the other side.

**Comment by GoldenBrownie** [comment]
It is updated. Maybe I'm just blind but I swear that option is nowhere to be found. Everything else works pretty great to me
